### PR TITLE
Compile as C

### PR DESCRIPTION
**Pull request by datalogics-staylor**
Specify library name and version:  **qr-code-generator/1.8.0**

* Compile as C when the package is built with the new "compile_as_c" option.

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jcar87** [comment]
Hi @datalogics-staylor , thanks for your contribution.
Is this a case where it might make sense for the package to just build _both_ libraries, C and C++? if the resulting library files have a different name, they can coexist, and then these can be different components? I suspect that's the case with other libraries in the same situation as well.

**Comment by datalogics-staylor** [comment]
Hi @jcar87. That's a very good point. I looked around a bit for recipes that do both C and C++ and didn't find many, so I chose to go this route so that it doesn't build packages that aren't required. It would be a pretty minor change to just build both, so I'd be happy to go that route if that seems to make more sense. Conan and CCI are relatively new to me, so I'm still learning common usages. Thanks for the feedback!

**Comment by uilianries** [comment]
@datalogics-staylor Your change looks correct in terms of CMake, but still need to update the Conanfile recipe in order to list the new library too. Could please update the pacakge_info as well?

File: recipes/qr-code-generator/all/conanfile.py
Comment (uilianries) [inline recipes/qr-code-generator/all/conanfile.py:106]:
```suggestion
        ]
        self.cpp_info.components["qrcodegencpp"].set_property("cmake_target_name", "qr-code-generator::qrcodegencpp")
        self.cpp_info.components["qrcodegencpp"].libs = ["qrcodegen" if Version(self.version) < "1.7.0" else "qrcodegencpp"]
        self.cpp_info.components["qrcodegenc"].set_property("cmake_target_name", "qr-code-generator::qrcodegenc")
        self.cpp_info.components["qrcodegenc"].libs = ["qrcodegenc"]
```

I would suggest adding separated components too, so you can link to C++ or C only, by using cmake targets.

Comment (datalogics-staylor) [reply recipes/qr-code-generator/all/conanfile.py:106]: Ah, that makes sense. I'm pretty new to both Conan and CMake, so I appreciate the feedback!

**Review by uilianries — APPROVED** [review]
Looks good. Both C++ and C libraries are available as separated targets.

**Review by toge — APPROVED** [review]

**Comment by datalogics-robb** [comment]
@franramirez688 We're just waiting for one more review - the last approval was 2 weeks ago - can we get some eyes on this one for us?

Thanks!

**Review by jcar87 — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**
